In [1]:
import pandas as pd


In [2]:
data = pd.read_csv('sample_3m.csv')

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000000 entries, 0 to 2999999
Data columns (total 30 columns):
 #   Column               Dtype  
---  ------               -----  
 0   property_id          int64  
 1   property_capacity    int64  
 2   property_address     object 
 3   booking_id           int64  
 4   check_in_date        object 
 5   check_out_date       object 
 6   base_fee             int64  
 7   service_fee          float64
 8   cleaning_fee         float64
 9   gst                  float64
 10  total_price          float64
 11  owner_id             int64  
 12  abn                  object 
 13  tin                  float64
 14  dob                  object 
 15  contact_number       int64  
 16  email                object 
 17  rating               int64  
 18  owner_location       object 
 19  owner_address        object 
 20  bank_account_name    object 
 21  bsb                  float64
 22  bank_account_number  float64
 23  bank_name            object 
 24

## Property Information

**property_id**: The idendification number for each regested property. (Mock)

**property_capacity**: What kind of group this property can hold. (Mock)

**property_address**: Real world postcode and its corresponding state and name.

## Order Information

**booking_id**: Non-unique booking id, this can be made unique or non-unique for test purpose.

**check_in_date**: Date between 01/01/23 - 30/06/23.

**check_out_date**: Date after check_in_date with 1-15 days more.

**base_fee**: Random number.

**service_fee**: 15% based on base_fee.

**cleaning_fee**: Random proportion based on base_fee.

**gst**: 10% based on base+service+cleaning_fee. 

**total_price**: gst * 110%.

## Owner Information

**owner_id**: Non-unique booking id, this can be made unique or non-unique for test purpose, 4-5 digits.

**abn**: 11 digits number in ABN's format, 80% chance of having this.

**tin**: 9 digits number if ABN is missing.

**dob**: Random date between 01/01/1940 - 31/12/1999.

**contact_number**: All contact_numbers are in Aussie style (+614*********).

**email**: first_name.last_name@gmail.com.

**rating**: Integer between 0 - 10.

### 80% chance inside Australia
* **owner_location**: Australia in this case.
* **owner_address**: Real world postcode and its corresponding state and name.
* **bank_account_name**: Fake name.
* **bsb**: 062***.
* **bank_account_number**: 1*** ****.
* **bank_name**: Commonwealth Bank of Australia in this case.
* **paypal_id**: 45% chance of having first_name.last_name@paypal.com, 45% having contact_number@paypal.com, 10% having empty.

### 20% chance in New Zealand
* **owner_location**: New Zealand in this case.
* **owner_address**: Auckland, NZ in this case.
* **bank_name**: ANZ Bank in this case.
* **iban**: AU********.
* **swift_code**: CTBAAU2S in this case.
  
## Order Status

**order_status**: 80% Completed, 8% Cancelled, 8% Refunded, 4% Parcially Refunded.

**final_price**: Completed: 100% totoal_price, Cancelled/Refunded: 0% totoal_price, Parcially Refunded: 50%-90% total_price.

**final_gst**: final_price / 11.